### ISSUE TITLE
Brave Shields Up/Down should use permission lifetimes

### ISSUE THREAD

**Issue by ShivanKaul** [brave/brave-browser#31233]
We do [fine-grained permissions per-site](https://brave.com/privacy-updates/8-grab-bag-2/). We should do something similar for Brave Shields, when the user switches off Shields i.e. let users say "turn down until I close tab/X mins/etc". 

UX TBD.

**Comment by bradleyrichter** [comment]
Many users quit Brave completely when they hit a broken site before they learn about shields to solve the broken site. For user who do find shields, if we reset shields too quickly,  those users would call it broken and then leave brave for good as well.

Thus I am not sure this change would actually retain more users. It would primarily retain more protections.

**Comment by bbondy** [comment]
Agreed. I'd also say it's more for user protection. I'll also highlight that I don't think it will hurt retention because this is a user that already knows how to turn down shields

**Comment by ShivanKaul** [comment]
I'd argue that this can help with user retention as well, since browsing a website with Shields Down results in the user seeing ads and annoyances and slow loading times etc, so an inferior product experience overall. Anecdotally (from helping debug webcompat issues), a lot of users simply forget that they'd turned down Shields previously for a site; I definitely do. 

I agree though that resetting Shields too quickly would be frustrating.
